### PR TITLE
741 oauth2

### PR DESCRIPTION
**Pull request by jenarp**

### CODE REVIEW

File: src/test/java/de/bonndan/nivio/security/LoginControllerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        LoginController loginController = new LoginController(linkFactory, properties);
+
+        // when
+        when(model.getAttribute("brandingLogoUrl")).thenReturn("nivio.icons.example.svg");
+        when(model.getAttribute("brandingForeground")).thenReturn("#FFFFFF");
+        when(model.getAttribute("brandingBackground")).thenReturn("#FFFFFF");
+        when(model.getAttribute("brandingMessage")).thenReturn("foo");
+
+        // then
+        assertThat(loginController.showLoginPage(model)).isEqualTo("login");
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/security/LoginControllerUnitTest.java:44]: You're effectively testing/mocking getters and setters of the view model. Instead, values should be assigned to the properties and tested on the model.

File: src/test/java/de/bonndan/nivio/security/LoginControllerCaseRequiredTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private MockMvc mvc;
+
+    @Value("${nivio.loginMode}")
+    private String loginMode;
+
+    @Test
+    void testShowLoginPageForRequired() throws Exception {
+
+
+        if (loginMode.equalsIgnoreCase(SecurityConfig.LOGIN_MODE_REQUIRED)) {
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/security/LoginControllerCaseRequiredTest.java:28]: If the annotation above works correctly, the if-clause should not be necessary.

File: src/test/java/de/bonndan/nivio/security/LoginControllerCaseOptionalTest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private MockMvc mvc;
+
+    @Value("${nivio.loginMode}")
+    private String loginMode;
+
+
+    @Test
+    void testShowLoginPageForOptional() throws Exception {
+
+        if (loginMode.equalsIgnoreCase(SecurityConfig.LOGIN_MODE_REQUIRED)) {
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/security/LoginControllerCaseOptionalTest.java:28]: remove if clause

File: src/test/java/de/bonndan/nivio/security/LoginControllerCaseNoneTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    @Autowired
+    private MockMvc mvc;
+
+    @Value("${nivio.loginMode}")
+    private String loginMode;
+
+    @Test
+    void testShowLoginPageForNone() throws Exception {
+
+        if (loginMode.equalsIgnoreCase(SecurityConfig.LOGIN_MODE_REQUIRED)) {
```
Comment (bonndan) [inline src/test/java/de/bonndan/nivio/security/LoginControllerCaseNoneTest.java:27]: remove if clause

File: src/main/resources/templates/login.html [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+<div class="container">
+    <div class="nivio-logo"><img src="/icons/svg/nivio.svg" alt="nivio logo" width="80" height="80"></div>
+    <div class="welcome"><h2 th:text="${brandingMessage}"></h2><h3>please login</h3><br></div>
+    <div class="brand-logo">
+        <a th:if="${auth_github} != null" th:href="@{(${auth_github.href})}">
+            <img src="/icons/svg/github.svg" alt="github logo" width="65" height="65">
+        </a>
+    </div>
+    <div class="brand-title">GITHUB</div>
```
Comment (bonndan) [inline src/main/resources/templates/login.html:67]: I forgot to move this inside the th:if blockl

**Review by bonndan — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/login.html (hunk 5 above)
Comment (jenarp) [reply src/main/resources/templates/login.html:67]: I do not understand 100 %, maybe we can talk tomorrow about it.

**Pull request merged**